distro/rpm/openvpn.init.d.rhel

```diff
@@ -224,17 +224,24 @@ case "$1" in
 	fi
 	;;
   status)
-	if [ -f $lock ]; then
-	    for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
-		if [ -s $pidf ]; then
-		    kill -USR2 `cat $pidf` >/dev/null 2>&1
-		fi
-	    done
-	    echo "Status written to /var/log/messages"
-	else
-	    echo "openvpn: service not started"
-	    exit 1
-	fi
+        if [ -f $lock ]; then
+            for pidf in `/bin/ls $piddir/*.pid 2>/dev/null`; do
+                PID=$(cat $pidf)
+                if [ -e /proc/$PID ]; then
+                    echo "openvpn is running. pid: $PID"
+                    exit 0
+                else
+                    echo "openvpn is dead but lock file exists: $lock and pid file exists: $pidf"
+                    exit 1
+                fi
+            done 
```

Comment:
Throwing several OpenVPN config files into /etc/openvpn and having the init script launch them all seems to be a common practice/use-case on both RedHat and Debian derivatives; the support is there on CentOS 6 and Debian 7.  While I agree that having a separate init script for each connection would be cleaner, but without proper configuration management system (e.g. puppet, ansible, chef or salt) setting them up would be a pain.

Fortunately the init script has been superceded by per-connetion services (e.g. openvpn@work, openvpn@home) on CentOS 7 and Debian 8, which use systemd. Interestingly none of the distros seem to use _our_ unit files, and have invented their own, which all seem to differ.

Anyways, how about returning 0 if all OpenVPN connections are fine, and returning 1 if _any_ OpenVPN connection has failed? Does that sound reasonable?